### ISSUE TITLE
value_template issue

### ISSUE THREAD

**Issue by blusius** [zigbee2mqtt/hassio-zigbee2mqtt#452]
### Description of the issue

Hi, I am getting constant error messages in HA logs. The ‘value_template’ option has been removed” errors

Devices becomes unavailable in HA but everything works in zigbee2mqtt itself.

I stopped z2m addon in HA and cleaned all messages from mqtt (was empty no retained messages on connection or somthing)
Started z2m and again messages apear with value_template under homeassistant/light/{device_id}/light/config

`{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"command_topic":"zigbee2mqtt/OfficeLight/set","device":{"identifiers":["zigbee2mqtt_0x54ef4410000fc31e"],"manufacturer":"Xiaomi","model":"Aqara smart wall switch H1 EU (no neutral, single rocker) (WS-EUK01)","name":"OfficeLight"},"json_attributes_topic":"zigbee2mqtt/OfficeLight","name":"OfficeLight","payload_off":"OFF","payload_on":"ON","state_topic":"zigbee2mqtt/OfficeLight","unique_id":"0x54ef4410000fc31e_light_zigbee2mqtt","value_template":"{{ value_json.state }}"}`

No value_template in devices.yaml at all. I have no ideas how to solve. seems that z2m on start sending bad discovery payload or somthing.  I edited those payloads on some light by removing value_template and in HA devices again apears. but after restart the system again failures in HA logs and devices unavailable.
Please help, I starting to give up and reinstall everything.

### Addon version

1.30.0-1

### Platform

 Home Assistant 2023.2.0 
Supervisor 2023.01.1 
Operating System 9.5 
Frontend 20230201.0 - latest 

### Logs of the issue (if applicable)

Exception in async_discover when dispatching 'mqtt_discovery_new_light_mqtt': ({'availability': [{'topic': 'zigbee2mqtt/bridge/state', 'value_template': '{{ value_json.state }}'}], 'command_topic': 'zigbee2mqtt/BathRoomLight/right/set', 'device': {'identifiers': ['zigbee2mqtt_0x54ef4410002460aa'], 'manufacturer': 'Xiaomi', 'model': 'Aqara smart wall switch H1 EU (no neutral, double rocker) (WS-EUK02)', 'name': 'BathRoomLight'}, 'json_attributes_topic': 'zigbee2mqtt/BathRoomLight', 'name': 'BathRoomLight right', 'payload_off': 'OFF', 'payload_on': 'ON', 'state_topic': 'zigbee2mqtt/BathRoomLight', 'state_value_template': '{{ value_json.state_right }}', 'unique_id': '0x54ef4410002460aa_light_right_zigbee2mqtt', 'value_template': '{{ value_json.state_right }}', 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 818, in validate_callable return schema(data) File "/usr/src/homeassistant/homeassistant/helpers/config_validation.py", line 829, in validator raise vol.Invalid(warning % arguments) voluptuous.error.Invalid: The 'value_template' option has been removed, please remove it from your configuration During handling of the above exception, another exception occurred: Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 334, in async_discover config: DiscoveryInfoType = discovery_schema(discovery_payload) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 232, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 355, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 351, in _exec v = func(v) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 818, in validate_callable return schema(data) File "/usr/src/homeassistant/homeassistant/components/mqtt/light/__init__.py", line 44, in validate_mqtt_light_discovery config: ConfigType = schemas[config_value[CONF_SCHEMA]](config_value) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 232, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 355, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 351, in _exec v = func(v) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 276, in __call__ raise er.MultipleInvalid([e]) voluptuous.error.MultipleInvalid: The 'value_template' option has been removed, please remove it from your configuration

**Comment by blusius** [comment]
There how it looks one of failing device  in devices.yaml
`'0x54ef4410002460aa':
  friendly_name: BathRoomLight
  homeassistant:
    switch:
      type: light
      object_id: light
    switch_left:
      type: light
      object_id: light_left
    switch_right:
      type: light
      object_id: light_right
    light:
      state_value_template: '{{ value_json.state }}'
    light_left:
      state_value_template: '{{ value_json.state_left }}'
    light_right:
      state_value_template: '{{ value_json.state_right }}'
  optimistic: true
  retain: false`

dont know who to blame. Seems the isue is on z2m side.

**Comment by blusius** [comment]
Update: I found how to fix it myself.
1. on light devices in devices.yaml as old one visible in previous post I added value_template: null and dicovery payload become as expected (without value_template)
```
light_right:
      value_template: null
      state_value_template: '{{ value_json.state_right }}'
```

strange that z2m cannot update devices itself and fix it, when it is known that it breaks HA and remove this frustration from user.  I expected that updating z2m would fix it. so maybe its "by design" or somehow failed to update properly

2. was issue that those lights somehow got new ids in HA and I missed that thinking that it is not discovered due to discovery errors. There I have no explanation and whos foult it is.

for example : `light.office_light` changed to `light.officelight`

But now everything managed. just only 2 days of pain!
So
I suggest z2m to do some checks over configuration or update it or suggest changes in UI if it is known that break environment.
Ok good luck.
